### PR TITLE
Fixing broken example tests

### PR DESCRIPTION
**Pull request by aaltay**
Nose does not pick up decorated tests unless the decorator name
starts with test_. This resulted in some test being inadvertently
disabled.

Also OptionsContext
(https://github.com/aaltay/incubator-beam/blob/python-sdk/sdks/python/apache_beam/utils/options.py#L457)
is not doing what it is supposed to do. Its augment_options() method is
not called therefore using OptionsContext does not override the option
values as expected. I will remove this class and a few uses of it in
tests, in a follow up.

Finally removed contains_in_any_order in favor of equal_to. (As
discuessed in:
https://github.com/apache/incubator-beam/pull/650#discussion_r70906340

### CODE REVIEW

**Comment by aaltay** [comment]
R: @chamikaramj 

Please review. Thank you for noticing these problems. I reverted the changes in group_with_coder_test.

File: sdks/python/apache_beam/examples/complete/estimate_pi_test.py [hunk 1]
```diff
@@ -47,5 +47,4 @@ def test_basics(self):
 
 
 if __name__ == '__main__':
-  logging.getLogger().setLevel(logging.INFO)
```
Comment (chamikaramj) [inline sdks/python/apache_beam/examples/complete/estimate_pi_test.py]: Why was this removed ? Shouldn't we print INFO logs by default ?

**Comment by chamikaramj** [comment]
LGTM

File: sdks/python/apache_beam/examples/complete/estimate_pi_test.py (hunk 1 above)
Comment (aaltay) [inline sdks/python/apache_beam/examples/complete/estimate_pi_test.py]: There were some `logging.info()` calls that was removed earlier. importing logging module become unnecessary at that point but I forgot to remove this call. We can keep this, to see info data from the sdk. What would you prefer?

Comment (chamikaramj) [inline sdks/python/apache_beam/examples/complete/estimate_pi_test.py]: I prefer INFO to be the default here and elsewhere. Some tests might print useful information. For example I'm printing INFO logs in source test framework - https://github.com/apache/incubator-beam/pull/667/files

I think if we don't set it will default to WARN.

Comment (aaltay) [inline sdks/python/apache_beam/examples/complete/estimate_pi_test.py]: Done. You are correct, the default is warning.

**Comment by robertwb** [comment]
This PR was merged, can be closed.

**Comment by aaltay** [comment]
Thank you.
